src/main/java/com/synopsys/integration/alert/common/descriptor/Descriptor.java

```diff
@@ -80,18 +82,27 @@ public void addDistributionUiConfig(final DescriptorActionApi descriptorActionAp
         addDistributionActionApi(descriptorActionApi);
     }
 
-    // TODO since global config is optional this should return an optional.  The lookups in the maps should return optionals since they could be null.
-    public DescriptorActionApi getRestApi(final ConfigContextEnum actionApiType) {
-        return restApis.get(actionApiType);
+    public Optional<DescriptorActionApi> getActionApi(final ConfigContextEnum actionApiType) {
+        return Optional.ofNullable(actionApis.get(actionApiType));
     }
 
-    public UIConfig getUIConfig(final ConfigContextEnum actionApiType) {
-        return uiConfigs.get(actionApiType);
+    public Optional<UIConfig> getUIConfig(final ConfigContextEnum actionApiType) {
+        return Optional.ofNullable(uiConfigs.get(actionApiType));
     }
 
-    public List<UIConfig> getAllUIConfigs() {
+    public Optional<DescriptorMetadata> getMetaData(final ConfigContextEnum context) {
+        final Optional<UIConfig> uiConfig = getUIConfig(context);
+
+        if (uiConfig.isEmpty()) {
+            return Optional.empty();
+        }
+
+        return Optional.of(createMetaData(uiConfig.get(), context));
```

Comment:
If we are cleaning up optional use based on the other comments then this should be cleaned up with the ifPresent(....) method

Reply:
Or something like: `return getUIConfig(context).map(uiConfig -> createMetaData(uiConfig, context));`

Reply:
ifPresent returns void so we would need to do something like mapping instead

Reply:
Gavin's suggestion is the better option in this case